Sources/PrimerSDK/Classes/User Interface/TokenizationViewModels/ApayaTokenizationViewModel.swift

```diff
@@ -425,12 +421,41 @@ extension ApayaTokenizationViewModel: WKNavigationDelegate {
 
 extension ApayaTokenizationViewModel {
     
-    override func handle(error: Error) {
-        ClientSession.Action.unselectPaymentMethod(resumeHandler: nil)
+    private func executeCompletionAndNullifyAfter(error: Error? = nil) {
         self.completion?(nil, error)
         self.completion = nil
     }
     
+    private func selectPaymentMethodWithParameters(_ parameters: [String: Any]) {
+        
+        firstly {
+            ClientSession.Action.selectPaymentMethodWithParameters(parameters)
+        }
+        .done {}
+        .catch { error in
+            self.handleErrorBasedOnSDKSettings(error)
+        }
+    }
+        
+    private func unselectPaymentMethodWithError(_ error: Error) {
+        firstly {
+            ClientSession.Action.unselectPaymentMethod()
+        }
+        .done {}
+        .catch { error in
+            self.handleErrorBasedOnSDKSettings(error)
+        }
+    }
+}
+
+extension ApayaTokenizationViewModel {
+    
+    override func handle(error: Error) {
+        self.executeCompletionAndNullifyAfter(error: error)
```

Comment:
This leads to the superclass completion getting called, which is going to call either `handleFailedTokenizationFlow`. This is also called 2 lines below.

This is done from other view models as well, please check them all.